Support storing nsrr_token in config or environment variable (#243)
cbrnr/sleepecg: docs/datasets.md

mesa = read_mesa()  # note that this is a generator
```

> As an alternative to always using the 'set_nsrr_token' function, the NSRR token can be set either in the config file
> (nsrr_token) or as an environment variable (NSRR_TOKEN). How to set an environment variable depends on your operating system and should be looked up in a relevant manual.

Reviewer: ```suggestion
Instead of always using [`set_nsrr_token()`](sleepecg.set_nsrr_token), you can set the NSRR token via [`set_config(nsrr_token="YOUR_TOKEN")`](sleepecg.set_config) or as an environment variable (`NSRR_TOKEN`).
```

> Setting the NSRR token in the config file is done via  [`config.py`](https://github.com/cbrnr/sleepecg/blob/main/src/sleepecg/config.py).
> The following code snippet shows how the set the NSRR token in the config file:
> 
> ```python
> from sleepecg.config import set_config 
> set_config(nsrr_token="your_nsrr_token")
> ```

Reviewer: ```suggestion
```

> Since the NSRR token can be set in multiple ways, a priority list is defined as such:
> 1. NSRR token set via [`set_nsrr_token()`][sleepecg.set_nsrr_token]
> 2. NSRR token set as an environment variable
> 3. NSRR token set in the config
> 
> So if for instance the NSRR token was set both via 1. and 3. the NSRR token set via 1. is used.

Reviewer: ```suggestion
SleepECG checks for the NSRR token in the following order:

1. Token set via [`set_nsrr_token()`][sleepecg.set_nsrr_token]
2. Token set via environment variable `NSRR_TOKEN`
3. Token set in the user configuration

For example, if the token is set by both method 1 *and* method 3, method 1 takes precedence.
```
